testify/test_rerunner.py

```diff
@@ -26,3 +38,35 @@ def discover(self):
 
             cls = test_discovery.import_test_class(module_path, class_name)
             yield self._construct_test(cls, name_overrides=methods)
+
+    def discover_json(self):
+        test = '{' + next(self.rerun_test_file)
+        prev_test = self._parse_json_test(test)
+        methods = [prev_test.method]
+        for test in self.rerun_test_file:
+            test = self._parse_json_test(test)
+            if test == prev_test._replace(method=test.method):
+                methods.append(test.method)
+            else:
+                yield self._construct_json_test(prev_test.module, prev_test.cls, methods)
+                prev_test = test
+                methods = [prev_test.method]
+        yield self._construct_json_test(prev_test.module, prev_test.cls, methods)
```

Comment:
These dozen lines are a mess.
I'd like to simplify while preserving the as-you-go aspect.